File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int res = INT_MIN;
        int i = 0, sum = 0;
        while( i < n){
            sum += nums[i];
            
            res = max(res, sum);
            
            if(sum < 0)
                sum = 0;
            i++;
        }
        return res;
    }
};

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int i = 0, j = 0, n = nums.length, maxSubarraySum = Integer.MIN_VALUE, currSubArraySum = 0;
        while(j < n){
            currSubArraySum += nums[j];
            maxSubarraySum = Math.max(currSubArraySum, maxSubarraySum);

            while(currSubArraySum < 0 && i <= j){
                currSubArraySum -= nums[i];
                i++;
            }
            j++;
        }
        return maxSubarraySum;
    }
}